client/src/main/java/org/spine3/change/MessageMismatch.java

```diff
@@ -70,6 +70,22 @@ public static ValueMismatch expectedNotDefault(Message expected, int version) {
     }
 
     /**
+     * Creates a {@code ValueMismatch} for a command that wanted to <em>change</em> a field value,
+     * but discovered that the field has the default value.
+     *
+     * @param expected the value expected by the command
+     * @param newValue the value the command wanted to set
+     * @param version  the version of the entity in which the mismatch is discovered
+     * @return new {@code ValueMismatch} instance
+     */
+    public static ValueMismatch expectedNotDefault(Message expected, Message newValue, int version) {
+        checkNotNull(expected);
```

Comment:
We annotate the package to say that the API expects non-nulls. But in the `public` and `protected` API we check for `null`s to prevent accidental errors and their propagation into our code deeper.